ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+        if (hMiraMonFeature)
+        {
+            hMiraMonFeature->nReadFeatures =
+                hMiraMonLayer->TopHeader.nElemCount - previusFID;
+        }
+        return re;
+    }
+    if (hMiraMonLayer->bIsDBF)
+    {
+        // Adding a record to DBF file
+        re = LOG_ACTION(MMCreateRecordDBF(hMiraMonLayer, hMiraMonFeature));
+        if (hMiraMonFeature)
+        {
+            hMiraMonFeature->nReadFeatures =
+                hMiraMonLayer->TopHeader.nElemCount - previusFID;
+        }
+        return re;
+    }
+
+    return MM_CONTINUE_WRITING_FEATURES;
+}
+
+/* -------------------------------------------------------------------- */
+/*      Tools used by MiraMon.                                          */
+/* -------------------------------------------------------------------- */
+
+void MMInitBoundingBox(struct MMBoundingBox *dfBB)
+{
+    if (!dfBB)
+        return;
+    dfBB->dfMinX = STATISTICAL_UNDEF_VALUE;
+    dfBB->dfMaxX = -STATISTICAL_UNDEF_VALUE;
+    dfBB->dfMinY = STATISTICAL_UNDEF_VALUE;
+    dfBB->dfMaxY = -STATISTICAL_UNDEF_VALUE;
+}
+
+void MMUpdateBoundingBox(struct MMBoundingBox *dfBBToBeAct,
+                         struct MMBoundingBox *dfBBWithData)
+{
+    if (!dfBBToBeAct)
+        return;
+
+    if (dfBBToBeAct->dfMinX > dfBBWithData->dfMinX)
+        dfBBToBeAct->dfMinX = dfBBWithData->dfMinX;
+
+    if (dfBBToBeAct->dfMinY > dfBBWithData->dfMinY)
+        dfBBToBeAct->dfMinY = dfBBWithData->dfMinY;
+
+    if (dfBBToBeAct->dfMaxX < dfBBWithData->dfMaxX)
+        dfBBToBeAct->dfMaxX = dfBBWithData->dfMaxX;
+
+    if (dfBBToBeAct->dfMaxY < dfBBWithData->dfMaxY)
+        dfBBToBeAct->dfMaxY = dfBBWithData->dfMaxY;
+}
+
+void MMUpdateBoundingBoxXY(struct MMBoundingBox *dfBB,
+                           struct MM_POINT_2D *pCoord)
+{
+    if (!pCoord)
+        return;
+
+    if (pCoord->dfX < dfBB->dfMinX)
+        dfBB->dfMinX = pCoord->dfX;
+
+    if (pCoord->dfY < dfBB->dfMinY)
+        dfBB->dfMinY = pCoord->dfY;
+
+    if (pCoord->dfX > dfBB->dfMaxX)
+        dfBB->dfMaxX = pCoord->dfX;
+
+    if (pCoord->dfY > dfBB->dfMaxY)
+        dfBB->dfMaxY = pCoord->dfY;
+}
+
+/* -------------------------------------------------------------------- */
+/*      Resize structures for reuse                                     */
+/* -------------------------------------------------------------------- */
+int MMResizeMiraMonFieldValue(struct MiraMonFieldValue **pFieldValue,
+                              MM_EXT_DBF_N_MULTIPLE_RECORDS *nMax,
+                              MM_EXT_DBF_N_MULTIPLE_RECORDS nNum,
+                              MM_EXT_DBF_N_MULTIPLE_RECORDS nIncr,
+                              MM_EXT_DBF_N_MULTIPLE_RECORDS nProposedMax)
+{
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nPrevMax;
+
+    if (nNum < *nMax)
+        return 0;
+
+    nPrevMax = *nMax;
+    *nMax = max_function(nNum + nIncr, nProposedMax);
+    if (((*pFieldValue) = realloc_function(
```

Comment:
the return value of realloc_function() should be set to a temporary variable, and checked for null, before assigning it to *pFieldValue. to avoid memory leaks in case of a realloc failure

Reply:
sure, done!